enzy_htp/_interface/pymol_interface.py

```diff
@@ -101,6 +102,111 @@ def select_pymol_obj(self, pattern: str, pymol_obj_name: str,
 
         return sorted(result)
 
+    def point_mutate(self, pos_key: Tuple[str, int],
+                     target: str, pymol_obj_name: str,
+                     pymol_session: pymol2.PyMOL,
+                     debug: bool = False) -> None:
+        """
+        Performs a single point mutation on the target WT in the PyMOL session in-place.
+        Args:
+            pos_key: the chain id and residue index.
+            target: the target residue name (3 letters).
+            pymol_obj_name: the name of the target WT in PyMOL.
+            pymol_session: the target PyMOL session.
+            debug: prints all rotamer scores out for debugging purposes.
+        Returns:
+            None.
+        """
+
+        # load pymol wizard mutagenesis function
+        pymol_session.cmd.wizard("mutagenesis")
+        pymol_session.cmd.do("refresh_wizard")
+        
+        # select res idx to mutate to target
+        pymol_session.cmd.get_wizard().set_mode(target)
+        if pymol_obj_name == "":
+            pymol_session.cmd.get_wizard().do_select(f"{pos_key[0]}/{str(pos_key[1])}/")
+        else:
+            pymol_session.cmd.get_wizard().do_select(f"{pymol_obj_name}//{pos_key[0]}/{str(pos_key[1])}/")
+
+        # prints all rotamers and strains out; also saves each variation to a PDB file in scratch/.
+        # can use for debugging purposes
+        if debug:
+            for i in range(1, pymol_session.cmd.count_states() + 1):
+                pymol_session.cmd.wizard("mutagenesis")
+                pymol_session.cmd.do("refresh_wizard")
+                pymol_session.cmd.get_wizard().set_mode(target)
+                if pymol_obj_name == "":
+                    pymol_session.cmd.get_wizard().do_select(f"{pos_key[0]}/{str(pos_key[1])}/")
+                else:
+                    pymol_session.cmd.get_wizard().do_select(f"{pymol_obj_name}//{pos_key[0]}/{str(pos_key[1])}/")
+            
+                pymol_session.cmd.get_wizard().do_state(i)
+                pymol_session.cmd.frame(i)
+                pymol_session.cmd.get_wizard().apply()
+                self.export_pymol_obj(pymol_obj_name, pymol_session, rotamer=i)
+            return
+
+        pymol_session.cmd.get_wizard().apply()
+
+    def export_pymol_obj(self, pymol_obj_name: str, 
```

Comment:
a better name aligning with export_enzy_htp_stru could be export_pdb (since we are always exporting pymol obj but to different target formats)
I didn't suggest export_file because we may add some treatment that specific to PDB (e.g.: whether use retain order)
Maybe in some day we need more than just PDB (e.g.: mol2) we will consider making a very general one export_file and call it in export_pdb